nixos/modules/services/x11/display-managers/default.nix

```diff
@@ -64,22 +69,23 @@ let
 
       # Speed up application start by 50-150ms according to
       # http://kdemonkey.blogspot.nl/2008/04/magic-trick.html
-      rm -rf "$HOME/.compose-cache"
-      mkdir "$HOME/.compose-cache"
+      compose_cache="''${XCOMPOSECACHE:-$HOME/.compose-cache}"
+      mkdir -p "$compose_cache"
+      # To avoid accidentally deleting a wrongly set up XCOMPOSECACHE directory,
+      # defensively try to delete cache *files* only, following the file format specified in
+      # https://gitlab.freedesktop.org/xorg/lib/libx11/-/blob/master/modules/im/ximcp/imLcIm.c#L330
+      # sprintf (*res, "%s/%c%d_%03x_%08x_%08x", dir, _XimGetMyEndian(), XIM_CACHE_VERSION, (unsigned int)sizeof (DefTree), hash, hash2);
```

Comment:
Is this an upstream source line reference?

Reply:
Yes, it is a link to the function that constructs the file name, and the line with the format that the regex in the PR is based on.
Last update to this file is from 2 years ago, so at least it seems to be rather stable.

Reply:
Can you add the exact line? I have not found it on the line you have provided:

https://gitlab.freedesktop.org/xorg/lib/libx11/-/blob/master/modules/im/ximcp/imLcIm.c#L330

```c
static int _XimCachedFileName (
    const char *dir, const char *name,
    const char *intname, const char *encoding,
    uid_t uid, int isglobal, char **res, off_t *size)
```

Reply:
Yes, it was pointing to the function declaration to show that this is about the cache file names, but I can see how that can be confusing. Changed it to highlight the lines with the format directly. Thanks for reviewing!